### PR TITLE
Security considerations: Point into corr-clar-future

### PR DESCRIPTION
**Pull request by chrysn**
As per today's interim, this is all that's between this document and a WGLC.

By the time we're working in the WGLC comments, we can update the reference that now points into a PR to point to corr-clar.

### CODE REVIEW

File: draft-ietf-core-dns-over-coap.md [hunk 1]
```diff
@@ -499,6 +507,11 @@ harden against injecting spoofed responses.
 Consequently, it is of little concern to leverage the benefits of CoAP caching by setting the ID to
 0.
 
+General CoAP security considerations apply.
+Exceeding those in {{Section 11 of RFC7252}},
+the request patterns of DoC make it likely that long-lived security contexts are maintained:
+{{amp-0rtt}} goes into more detail on what can and needs to be done
+when those are resumed from a new address.
```
Comment (miri64) [inline draft-ietf-core-dns-over-coap.md:514]: With the opening "General CoAP security considerations apply." wouldn't it make more sense to make this the first paragraph?

File: draft-ietf-core-dns-over-coap.md [hunk 2]
```diff
@@ -499,6 +507,11 @@ harden against injecting spoofed responses.
 Consequently, it is of little concern to leverage the benefits of CoAP caching by setting the ID to
 0.
 
+General CoAP security considerations apply.
+Exceeding those in {{Section 11 of RFC7252}},
+the request patterns of DoC make it likely that long-lived security contexts are maintained:
+{{amp-0rtt}} goes into more detail on what can and needs to be done
```
Comment (miri64) [inline draft-ietf-core-dns-over-coap.md:513]:
```suggestion
{{amp-0rtt}} goes into more detail on what needs to be done
```

Unless I am missing something regarding IETF lingo, the "can" is superfluous.

File: draft-ietf-core-dns-over-coap.md (hunk 1 above)
Comment (miri64) [inline draft-ietf-core-dns-over-coap.md:514]:
```suggestion
when those are resumed from a new source address or port.
```

Comment (chrysn) [reply draft-ietf-core-dns-over-coap.md:514]: Maybe 'new endpoint' – source-address-and-port is a UDP/TCP thing. Thinking of OSCORE, you're also in a "new endpoint" situation if after requests over TCP from an address/port combination, all of a sudden the requests come from the same UDP address/port. (One might argue that it's very likely that this is return routable if TCP was, the same argument can also be made for same-IP-different-port, and then we'd have to think about NAT, and I don't want to think about NAT).

Comment (chrysn) [reply draft-ietf-core-dns-over-coap.md]: Moved to the top.

Comment (chrysn) [reply draft-ietf-core-dns-over-coap.md]: Updated to "new endpoint" – that fine?

File: draft-ietf-core-dns-over-coap.md [hunk 3]
```diff
@@ -77,7 +77,15 @@ informative:
   I-D.lenders-core-dnr: core-dnr
   I-D.amsuess-core-cachable-oscore: cachable-oscore
   DoC-paper: DOI.10.1145/3609423
-
+  amp-0rtt:
+   title: PR #40 "Amplification and 0-RTT" on "CoAP: Corrections and Clarifications"
```
Comment (miri64) [inline draft-ietf-core-dns-over-coap.md:81]:
```suggestion
   title: 'PR #40 "Amplification and 0-RTT" on "CoAP: Corrections and Clarifications"'
```

File: draft-ietf-core-dns-over-coap.md [hunk 4]
```diff
@@ -77,7 +77,15 @@ informative:
...
+   date: 2024-09-25
+   format:
+     HTML: https://github.com/core-wg/corrclar/pull/40
+   note: |
+     It is expected that that PR will be merged way ahead of this document's publication;
```
Comment (miri64) [inline draft-ietf-core-dns-over-coap.md:86]:
```suggestion
   ann: |
     Note: It is expected that that PR will be merged way ahead of this document's publication;
```

**Review by miri64 — CHANGES_REQUESTED** [review]

**Review by miri64 — APPROVED** [review]

**Pull request merged**